module/Finna/src/Finna/RecordDriver/SolrEad3.php

```diff
@@ -286,8 +310,12 @@ public function getUnitIds()
             if (!$label || !$val) {
                 continue;
             }
-            $label = $this->translate("Unit ID:$label");
-            $ids[$label] = $val;
+            $key = "Unit ID:$label";
+            $translated = $this->translate($key);
+            if ($translated !== $key) {
```

Comment:
translate:n pitäisi jo palauttaa tuolle avaimelle $label, jos ei käännöstä löydy. Turha tarkistus/muutos? Jos ei muuten toimi, niin translate:lle voi kertoa lisäparametrilla, mitä palautetaan oletuksena.

Reply:
Muutin käyttämään oletusta, paljon siistimpi näin.